Log partition movements before throwing exception

mhratson opened the pull request:
## Details

- Provide partition movement details in`OngoingExecutionException` exception message.
- Having partition movement details is handy for troubleshooting in cases when partition movements are ongoing for a long time, providing necessary details for potential mitigation.

CCisGG commented on cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java:
```diff
@@ -1048,15 +1049,19 @@ private void startExecution(LoadMonitor loadMonitor,
     try {
-      hasOngoingPartitionReassignments = hasOngoingPartitionReassignments();
+       ongoingPartitionReassignments = ExecutionUtils.ongoingPartitionReassignments(_adminClient);
+      // avoid calling hasOngoingPartitionReassignments() to save 1 request.
```
I think it's fine to remove this comment. This is probably only be useful during review time.

CCisGG commented on cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java:
```diff
@@ -1048,15 +1049,19 @@ private void startExecution(LoadMonitor loadMonitor,
     if (hasOngoingPartitionReassignments) {
-      throw new OngoingExecutionException("There are ongoing inter-broker partition movements.");
+      LOG.error("Ongoing inter-broker partition movements: {}", ongoingPartitionReassignments);
+      throw new OngoingExecutionException("There are ongoing inter-broker partition movements");
```
I think the error log here could be redundant. Ideally you only need this:

`throw new OngoingExecutionException("There are ongoing inter-broker partition movements: " + ongoingPartitionReassignments); `

And then the caller whoever handle that exception should be able to log error.

mhratson replied:
My concern is if there's alot of partition movements the exception message will be pretty big which is not exactly desirable output. 
But if that's acceptable then i'm cool with that. 
Will make updates.

CCisGG replied:
That should be fine. It will be logged either ways.

mhratson committed:
Add partition movement details to `OngoingExecutionException` exception

when doing inter-broker partition movements during sanityCheckOngoingMovement

Having partition movement details is handy for troubleshooting
in cases when partition movements are ongoing for a long time.
